modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/packageSpec.scala

```diff
@@ -197,4 +200,41 @@ class packageSpec extends FlatSpec {
       ).unsafePerformSync._2.get._2.sequences(seqId3).status === SequenceState.Running
     )
   }
+
+  "engine" should "keep processing input messages regardless of how long Actions take" in {
+    val q = async.boundedQueue[Event](10)
+    val startedFlag = new Semaphore(0)
+    val finishFlag = new Semaphore(0)
+
+    val qs = Engine.State(Conditions.default,
+      None,
+      Map((seqId, Sequence.State.init(Sequence(
+        "First",
+        SequenceMetadata("GMOS", None),
+        List(
+          Step(
+            1,
+            None,
+            config,
+            Set(Resource.GMOS),
+            breakpoint = false,
+            List(
+              List(Task.apply{
```

Comment:
This worries me a bit. The test would not work if I used `delay` instead of `apply`. Which means that the engine is not robust to the kind of Tasks passed as Actions.